www/src/content/blog/2024-12-03-cf-service-bindings.mdx

```diff
@@ -0,0 +1,94 @@
+Separately, and not as part of the registration process, you offer a newsletter subscription.
+When an athlete subscribes to the newsletter, their email address is stored in a different database, and they receive a welcome email.
+
+In this scenario, you could have three workers: one for handling the sign-up form, another for managing the newsletter subscription, and then a _shared_ email worker to send emails.
+![Example Workers architecture](@/assets/blog/2024-12-03-workers-example-architecture.png)
+
+### How it works
+
+When one Cloudflare Worker invokes another, the calls remain within the same execution thread.
+This design ensures that breaking down service logic into multiple Workers does not introduce network latency, making the architecture both flexible and high-performance.
+
+There are two primary ways Workers can communicate using Service Bindings:
+
+- RPC
+- HTTP
+
+Bindings are configured in the `wrangler.toml` file. If you want to learn more about setting up Service Bindings, check out the [Cloudflare documentation](https://developers.cloudflare.com/workers/runtime-apis/bindings/service-bindings/).
+Since the calls occur in the same thread, it’s important to handle them asynchronously.
+Otherwise, Worker A might terminate the thread before Worker B has completed its operation. Here is a simple example using rpc:
+
+```typescript
+await c.env.WORKER_EMAIL.send(email, firstName);
+```
+
+If the Service Binding is defined in the `wrangler.toml` file, the Worker can invoke another Worker’s function by using the binding name and the appropriate function from the other Worker.
+
+## Developing with multiple Workers
+
+As service distribution increases, so does the complexity of observing and debugging a system.
+Tracing the full call chain of a request becomes critical to understanding performance and error handling.
+This is particularly important when dealing with asynchronous background tasks executed by other Workers.
+
+Observing and measuring response times is already an important aspect of local development.
+It helps you understand your application’s performance and identify bottlenecks.
+
+## Example with Fiberplane
+
+Let’s explore how Fiberplane’s instrumentation client can help trace requests between Cloudflare Workers.
+
+In this example, we use Fiberplane's client to instrument the sign-up Worker, which stores user data in a database and invokes the email Worker to send a confirmation email.
+With Fiberplane, you can visualize the entire call chain of this request.
+
+![Sequential Trace](@/assets/blog/2024-12-03-workers-binding-sequential.png)
+
+In the trace above, we see a sequential execution model where the sign-up Worker first performs database operations and then invokes the email Worker.
+The total execution time depends on the entire call chain, as shown below.
+
+We can improve the response time by using parallel execution with `waitUntil()`. Using Fiberplane, we can also visualize the improved execution model:
```

Comment:
only note: would be good to have a code sample along with this, as it could be kind of confusing what you mean by "use waitUntil" without seeing some reference code